.github/workflows/ci.yaml

```diff
@@ -4,19 +4,14 @@ on:
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
-env:
-  UV_SYSTEM_PYTHON: true
 jobs:
   lint:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
-        with:
-          python-version: 3.x
-          check-latest: true
-          cache: 'pip'
-      - run: make deps
+      - uses: astral-sh/setup-uv@v3
```

Comment:
It appears this action has caching support. https://github.com/astral-sh/setup-uv?tab=readme-ov-file#enable-caching

Did you have a chance to try it?